crates/primitives/starknet/src/crypto/hash/mod.rs

```diff
@@ -20,19 +22,27 @@ impl ThreadSafeCopy for Hasher {}
 
 /// Implement the `HasherT` trait for the `Hasher` enum.
 impl HasherT for Hasher {
-    fn hash(&self, data: &[u8]) -> Felt252Wrapper {
+    fn hash_bytes(&self, data: &[u8]) -> Felt252Wrapper {
         match self {
-            Self::Pedersen(p) => p.hash(data),
-            Self::Poseidon(p) => p.hash(data),
+            Self::Pedersen(p) => p.hash_bytes(data),
+            Self::Poseidon(p) => p.hash_bytes(data),
         }
     }
 
-    fn hash_elements(&self, data: &[Felt252Wrapper]) -> Felt252Wrapper {
+    fn compute_hash_on_wrappers(&self, data: &[Felt252Wrapper]) -> Felt252Wrapper {
         match self {
```

Comment:
cant't we just do `self. compute_hash_on_wrappers` tho ?

Reply:
Isn't this not possible because of the `Hasher` variants' field is what we need to actually match against?